### PR TITLE
Update serenity

### PR DESCRIPTION
**Pull request by jepcd**
https://discord.com/channels/381880193251409931/919310428344029265/1030184268309200949

### CODE REVIEW

**Comment by kangalio** [comment]
Sorry, I misunderstood your question of "I just box it right?". Can you change the PR to keep the function signatures the same and instead dereference the boxes?

The philosophy in the serenity boxing PR is to require dealing with Box as little as possible for the user by only boxing fields, but passing around structs themselves unboxed.

**Comment by jepcd** [comment]
done

**Comment by kangalio** [comment]
Hi, could you fix the errors? (wrong timestamp method?) Not sure what the MSRV is on about, maybe it's fixed if we just raise poise MSRV on serenity-next to serenity's MSRV on next

Btw #114 and #116 are basically duplicates of each other... heh :sweat_smile: But Gnome isi currently not responsive on the other PR

**Comment by jepcd** [comment]
`ApplicationCommandInteraction` and `CreateApplicationCommandOption` don't seem to be in serenity anymore

**Comment by kangalio** [comment]
Thank you for working on this

**Comment by jepcd** [comment]
@kangalioo can you approve the workflow? it should be working now

**Comment by jepcd** [comment]
Also can I just say, I love all the new changes in serenity, especially the builders! :D

**Comment by jepcd** [comment]
Should be fixed now

File: macros/src/modal.rs [hunk 1]
```diff
@@ -65,22 +65,24 @@ pub fn modal(input: syn::DeriveInput) -> Result<TokenStream, darling::Error> {
...
+            //         // Option::from().unwrap_or_default() dance to handle both T and Option<T>
+            //         b = b.value(Option::from(default).unwrap_or_else(String::new));
+            //     }
+            //     b
+            //         #( .placeholder(#placeholder) )*
+            //         .required(#required)
+            //         #( .min_length(#min_length) )*
+            //         #( .max_length(#max_length) )*
+            // }))
         });
```
Comment (jepcd) [inline macros/src/modal.rs]: @kangalioo can you help with this macro? I can't get it to work

Comment (jepcd) [reply macros/src/modal.rs]: builders should basically now be a `Vec<CreateActionRow>`

Comment (jepcd) [reply macros/src/modal.rs]: I got the style label and id to work, but not the other fields

Comment (kangalio) [reply macros/src/modal.rs]: I attempted a fix

**Comment by kangalio** [comment]
Looking good, finally! Thanks for getting this rather chunky PR done @alakhpc

**Pull request merged**